src/Metadata/Property/Factory/InheritedPropertyNameCollectionFactory.php

```diff
@@ -41,7 +41,7 @@ public function create(string $resourceClass, array $options = []): PropertyName
 
         // Inherited from parent
         foreach ($this->decorated->create($resourceClass, $options) as $propertyName) {
-            $propertyNames[$propertyName] = true;
+            $propertyNames[$propertyName] = (string) $propertyName;
```

Comment:
mm no, this is this one in my case which was causing the issue : the symfony component returns an int instead of a string